tests/test_git/common.py

```diff
@@ -0,0 +1,60 @@
+import sublime
+import subprocess
+import os
+import shutil
+import tempfile
+from unittesting import DeferrableTestCase
+
+
+# copy from https://github.com/randy3k/ProjectManager/blob/f8667697a77a9e7020891e3227de25c2ebb990b3/pm.py#L72
+def subl(args=[]):
+    executable_path = sublime.executable_path()
+    if sublime.platform() == 'osx':
+        app_path = executable_path[:executable_path.rfind(".app/") + 5]
+        executable_path = app_path + "Contents/SharedSupport/bin/subl"
+    subprocess.Popen([executable_path] + list(args))
+    if sublime.platform() == "windows":
+        def fix_focus():
+            window = sublime.active_window()
+            view = window.active_view()
+            window.run_command('focus_neighboring_group')
+            window.focus_view(view)
+        sublime.set_timeout(fix_focus, 300)
+
+
+class GSCreateRepoTestCase(DeferrableTestCase):
+
+    _temp_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup a repo for testing
+        """
+        cls._temp_dir = tempfile.mkdtemp()
+        subl(["-n", cls._temp_dir])
+
+        def condition():
+            for d in sublime.active_window().folders():
+                # on Windows, `cls._temp_dir` is lowered cased,
+                # `os.path.normcase` is needed for comparison.
+                if cls._temp_dir == os.path.normcase(d):
+                    return True
+
+        yield condition
```

Comment:
Deferrable test cases are executed by the [DeferringTextTestRunner][1] and it expects not only regular test functions, but also generators. If the test function is a generator, it does the following

- if the yielded object is a callable, the runner will evaluate the callable and check its returned value. If the result is `True`, the runner continues the generator, if not, the runner will wait until the condition is met.

- If the yielded object is an integer, say `x`, then it will [continue][3] the generator after `x` ms.

- Otherwise, the `yield` statement will always wait for 10 ms.
  
[1]: https://github.com/randy3k/UnitTesting/blob/dc810ee334bb031710b859478faaf50293880995/unittesting/core/st3/runner.py#L7
[2]: https://github.com/randy3k/UnitTesting/blob/dc810ee334bb031710b859478faaf50293880995/unittesting/core/st3/runner.py#L49
[3]: https://github.com/randy3k/UnitTesting/blob/dc810ee334bb031710b859478faaf50293880995/unittesting/core/st3/runner.py#L57